src/lib/FieldTypeProcessorRegistry.php

```diff
@@ -32,9 +32,9 @@ public function __construct(array $processors = [])
      * Registers $processor for $fieldTypeIdentifier.
      *
      * @param string $fieldTypeIdentifier
-     * @param \EzSystems\EzPlatformRest\FieldTypeProcessor $processor
+     * @param \EzSystems\EzPlatformRest\FieldTypeProcessorInterface $processor
      */
-    public function registerProcessor($fieldTypeIdentifier, FieldTypeProcessor $processor)
+    public function registerProcessor($fieldTypeIdentifier, FieldTypeProcessorInterface $processor)
```

Comment:
Don't forget to replace phpdoc annotation for the `$this->processors` property and `__construct()` as well.

As for the BC break, we could drop the type for this method (which would cause it to remain backwards compatible, since it's entirely valid to narrow down type in implementations) and:
 * throw an `InvalidArgumentException` if an object is passed that neither is a `FieldTypeProcessor` nor implements `FieldTypeProcessorInterface`.
 * trigger deprecation notice if an object without `FieldTypeProcessorInterface` is passed to it (but extends `FieldTypeProcessor`).

We can leave the `FieldTypeProcessorInterface` commented out inline, so it can be "easily" enforced in 4.0.

Reply:
Can do. It feels a bit hackish but definitely a good way to evolve while maintaining BC.
Detail: I think throwing TypeError wiould be even better, as that's what is thrown atm and will be thrown when enforcing type via the sugnature

Reply:
Done (and squashed)

Reply:
Thinking more about it... now `FieldTypeProcessor` does implement `FieldTypeProcessorInterface`, so the case of the deprecation notice seems impossible